### PR TITLE
Add some missing getter functions to the lua API

### LINKED ISSUE #1496: Add more getter functions/methods for the Lua API
The Lua API has a lot of setter functions/methods (for setting a value) but it lacks many corresponding getter functions (for receiving a value). I suggest to add these getter functions.

I personally think there is in most cases no reason *not* to have such functions. It is just inconsistent and incomplete.

The suggested function/method names are like the setter functions, but with “get” instead of “set”. There is a single exception. After the arrow, I write the suggested return value. It is just the same as the input parameters. Remember that Lua functions can return multiple values at once, for example: `return 1,2,3,4`. I wrote the functions which I consider the most important in bold type.

So, here I list the functions I think are lacking in the Lua API:

For `ObjectRef`s:

* **`get_properties` → `object_property_table`**
* **`get_armor_groups` → `table_of_group_ratings`** (I tested it with the `luacmd` mod this function does *not* exist for players; since players are basically `ObjectRef`s, I conclude that this is true for other `ObjectRef`s as well)
* `get_animation` → `group_rating_table, frame_speed, frame_blend`
* `get_attach` → `parent, bone, position, rotation`
* `get_bone_position` → `bone, position, rotation`

For player objects:

* **`get_physics_override` → the same table used as input for `set_physics_override`**
* `hud_get_hotbar_itemcount` → number
* `hud_get_hotbar_image` → texture path
* `hud_get_hotbar_selected_image` → texture path
* `get_sky` → `bgcolor, type, {texture names}`
* `get_day_night_ratio` (getter function for to `override_day_night_ratio`) → ratio or `nil`
* `get_local_animation` → see input parameters of `set_local_animation`
* `get_eye_offset` → `first_person_offset, third_person_offset`

Global functions:
* `minetest.get_gen_notify` → flags (a `string`)
* `minetest.get_player_password` (OK, let’s just forget this suggestion, shall we?)
* `minetest.get_mapgen_params` → `MapgenParams`
* `minetest.get_noiseparam_defaults` → see `minetest.set_noiseparam_defaults`

For all suggested function names, I checked `lua_api.txt` if it already exists. It did not.

If you think I missed a function, please write a comment.

### PR DESCRIPTION
**Pull request by TeTpaAka**
Fixes #1496

Adds:
ObjectRef:
* get_properties
* get_armor_groups
* get_animation
* get_attach
* get_bone_position

Players:
* get_physics_override
* hud_get_hotbar_itemcount
* hud_get_hotbar_image
* hud_get_hotbar_selected_image
* get_sky
* get_day_night_ratio
* get_local_animation
* get_eye_offset

Global:
* minetest.get_gen_notify
* minetest.get_noiseparams (set_noiseparams_defaults got renamed)

Doesn't add:
* minetest.get_player_password
* minetest.get_mapgen_params => is already there

### CODE REVIEW

File: src/content_sao.cpp [hunk 1]
```diff
@@ -541,12 +546,25 @@ void LuaEntitySAO::setAnimation(v2f frame_range, float frame_speed, float frame_
...
+	*frame_blend = m_animation_blend;
+}
+
 void LuaEntitySAO::setBonePosition(std::string bone, v3f position, v3f rotation)
 {
 	m_bone_position[bone] = core::vector2d<v3f>(position, rotation);
 	m_bone_position_sent = false;
 }
 
+void LuaEntitySAO::getBonePosition(std::string bone, v3f *position, v3f *rotation)
```
Comment (est31) [inline src/content_sao.cpp]: Always better to have `const std::string &` as type, that will spare a copy.

File: src/player.h [hunk 2]
```diff
@@ -215,6 +215,57 @@ class Player
...
+	}
+	std::string getHotbarSelectedImage() {
+		return hud_hotbar_selected_image;
+	}
+
+	void setSky(const video::SColor &bgcolor, const std::string &type,
+			const std::vector<std::string> &params) {
+		m_sky_bgcolor = bgcolor;
+		m_sky_type = type;
+		m_sky_params.clear();
```
Comment (est31) [inline src/player.h]: Is this needed?

File: src/script/common/c_content.cpp [hunk 3]
```diff
@@ -896,6 +954,12 @@ u32 read_flags_table(lua_State *L, int table, FlagDesc *flagdesc, u32 *flagmask)
 	return flags;
 }
 
+void write_flags_string(lua_State *L, FlagDesc *flagdesc, u32 flags, u32 flagmask)
```
Comment (est31) [inline src/script/common/c_content.cpp]: Aren't these methods always called `push_something` ?

File: src/script/lua_api/l_object.cpp [hunk 4]
```diff
@@ -1273,6 +1500,26 @@ int ObjectRef::l_override_day_night_ratio(lua_State *L)
 	return 1;
 }
 
+// get_day_night_ration(self)
```
Comment (est31) [inline src/script/lua_api/l_object.cpp]: This should be **ratio**

**Comment by est31** [comment]
See comments, otherwise :+1:

File: src/player.h (hunk 2 above)
Comment (TeTpaAka) [inline src/player.h]: No. Fixed.

File: src/script/common/c_content.cpp (hunk 3 above)
Comment (TeTpaAka) [inline src/script/common/c_content.cpp]: Yes. Fixed.

File: src/script/lua_api/l_object.cpp (hunk 4 above)
Comment (TeTpaAka) [inline src/script/lua_api/l_object.cpp]: Typo. Fixed.

File: src/content_sao.cpp (hunk 1 above)
Comment (TeTpaAka) [inline src/content_sao.cpp]: It doesn't work for me. I tried it with https://github.com/Rui914/playeranim and PlayerSAO, but when I pass const std::string & the function returns only zeros. { x = 0, y = 0, z = 0 }. With std::string it returns the actual value.

Comment (TeTpaAka) [inline src/content_sao.cpp]: Fixed.

**Comment by est31** [comment]
Can you rebase this?

**Comment by TeTpaAka** [comment]
Rebased.

**Comment by est31** [comment]
c0335f7d13ee46c6a46d0ceebea96960439ec9fd

**Pull request closed without merging**